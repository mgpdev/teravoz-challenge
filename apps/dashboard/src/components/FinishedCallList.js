import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import Paper from "@material-ui/core/Paper";
import classNames from "classnames";
import Button from "@material-ui/core/Button";
import SaveIcon from "@material-ui/icons/Save";
import moment from "moment";
import { subscribeToQueue } from "./../services/web-socket";
import { formatPhoneNumber } from "./../utils/formatter";
import logger from "./../utils/logger";

const styles = theme => ({
  root: {
    width: "100%",
    marginTop: theme.spacing.unit * 3,
    overflowX: "auto"
  },
  table: {
    minWidth: 700
  },
  title: {
    margin: theme.spacing.unit * 3
  },
  button: {
    margin: theme.spacing.unit
  },
  leftIcon: {
    marginRight: theme.spacing.unit
  },
  iconSmall: {
    fontSize: 20
  }
});

class FinishedCallList extends React.Component {
  constructor(props) {
    super(props);
    this.state = { calls: [] };
    subscribeToQueue("999", calls => {
      logger("Event (999) Received", calls);
      this.setState({
        calls: JSON.parse(calls)
      });
    });
  }
  render() {
    const { classes } = this.props;
    return (
      <Paper className={classes.root}>
        <Typography variant="title" gutterBottom className={classes.title}>
          Last finished Calls
        </Typography>
        <Table className={classes.table}>
          <TableHead>
            <TableRow>
              <TableCell>Number</TableCell>
              <TableCell>Start Time</TableCell>
              <TableCell>End Time</TableCell>
              <TableCell>Download</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {this.state.calls.map(row => {
              return (
                <TableRow key={row.id}>
                  <TableCell>{formatPhoneNumber(row.number)}</TableCell>
                  <TableCell>
                    {moment(row.startAt).format("YYYY-MM-DD HH:mm")}
                  </TableCell>
                  <TableCell>
                    {moment(row.endAt).format("YYYY-MM-DD HH:mm")}
                  </TableCell>
                  <TableCell>
                    {row.downloadLink && (
                      <Button
                        variant="contained"
                        size="small"
                        className={classes.button}
                      >
                        <SaveIcon
                          className={classNames(
                            classes.leftIcon,
                            classes.iconSmall
                          )}
                        />
                        Save
                      </Button>
                    )}
                  </TableCell>
                </TableRow>
              );
            })}
          </TableBody>
        </Table>
      </Paper>
    );
  }
}

FinishedCallList.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(FinishedCallList);
