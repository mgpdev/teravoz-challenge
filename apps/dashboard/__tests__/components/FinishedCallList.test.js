import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

// Component to be tested
import FinishedCallList from "./../../src/components/FinishedCallList";

describe("<FinishedCallList />", () => {
  describe("render()", () => {
    test("renders the component", () => {
      const wrapper = shallow(<FinishedCallList />);

      const component = wrapper.html();

      expect(component).toMatchSnapshot();
    });
  });
});
