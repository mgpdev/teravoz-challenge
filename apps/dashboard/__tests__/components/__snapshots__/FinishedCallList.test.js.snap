// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FinishedCallList /> render() renders the component 1`] = `"<div class=\\"MuiPaper-root-4 MuiPaper-elevation2-8 MuiPaper-rounded-5 FinishedCallList-root-1\\"><h2 class=\\"MuiTypography-root-31 MuiTypography-title-37 MuiTypography-gutterBottom-48 FinishedCallList-title-3\\">Finished Calls</h2><table class=\\"MuiTable-root-56 FinishedCallList-table-2\\"><thead class=\\"MuiTableHead-root-57\\"><tr class=\\"MuiTableRow-root-58 MuiTableRow-head-61\\"><th class=\\"MuiTableCell-root-63 MuiTableCell-head-64\\" scope=\\"col\\">Customer</th><th class=\\"MuiTableCell-root-63 MuiTableCell-head-64 MuiTableCell-numeric-67\\" scope=\\"col\\">Number</th><th class=\\"MuiTableCell-root-63 MuiTableCell-head-64 MuiTableCell-numeric-67\\" scope=\\"col\\">Start Time</th><th class=\\"MuiTableCell-root-63 MuiTableCell-head-64 MuiTableCell-numeric-67\\" scope=\\"col\\">End Time</th><th class=\\"MuiTableCell-root-63 MuiTableCell-head-64 MuiTableCell-numeric-67\\" scope=\\"col\\">Download</th></tr></thead><tbody class=\\"MuiTableBody-root-71\\"><tr class=\\"MuiTableRow-root-58\\"><th class=\\"MuiTableCell-root-63 MuiTableCell-body-65\\" scope=\\"row\\">Frozen yoghurt</th><td class=\\"MuiTableCell-root-63 MuiTableCell-body-65 MuiTableCell-numeric-67\\">159</td><td class=\\"MuiTableCell-root-63 MuiTableCell-body-65 MuiTableCell-numeric-67\\">6</td><td class=\\"MuiTableCell-root-63 MuiTableCell-body-65 MuiTableCell-numeric-67\\">24</td><td class=\\"MuiTableCell-root-63 MuiTableCell-body-65 MuiTableCell-numeric-67\\">4</td></tr><tr class=\\"MuiTableRow-root-58\\"><th class=\\"MuiTableCell-root-63 MuiTableCell-body-65\\" scope=\\"row\\">Ice cream sandwich</th><td class=\\"MuiTableCell-root-63 MuiTableCell-body-65 MuiTableCell-numeric-67\\">237</td><td class=\\"MuiTableCell-root-63 MuiTableCell-body-65 MuiTableCell-numeric-67\\">9</td><td class=\\"MuiTableCell-root-63 MuiTableCell-body-65 MuiTableCell-numeric-67\\">37</td><td class=\\"MuiTableCell-root-63 MuiTableCell-body-65 MuiTableCell-numeric-67\\">4.3</td></tr><tr class=\\"MuiTableRow-root-58\\"><th class=\\"MuiTableCell-root-63 MuiTableCell-body-65\\" scope=\\"row\\">Eclair</th><td class=\\"MuiTableCell-root-63 MuiTableCell-body-65 MuiTableCell-numeric-67\\">262</td><td class=\\"MuiTableCell-root-63 MuiTableCell-body-65 MuiTableCell-numeric-67\\">16</td><td class=\\"MuiTableCell-root-63 MuiTableCell-body-65 MuiTableCell-numeric-67\\">24</td><td class=\\"MuiTableCell-root-63 MuiTableCell-body-65 MuiTableCell-numeric-67\\">6</td></tr><tr class=\\"MuiTableRow-root-58\\"><th class=\\"MuiTableCell-root-63 MuiTableCell-body-65\\" scope=\\"row\\">Cupcake</th><td class=\\"MuiTableCell-root-63 MuiTableCell-body-65 MuiTableCell-numeric-67\\">305</td><td class=\\"MuiTableCell-root-63 MuiTableCell-body-65 MuiTableCell-numeric-67\\">3.7</td><td class=\\"MuiTableCell-root-63 MuiTableCell-body-65 MuiTableCell-numeric-67\\">67</td><td class=\\"MuiTableCell-root-63 MuiTableCell-body-65 MuiTableCell-numeric-67\\">4.3</td></tr><tr class=\\"MuiTableRow-root-58\\"><th class=\\"MuiTableCell-root-63 MuiTableCell-body-65\\" scope=\\"row\\">Gingerbread</th><td class=\\"MuiTableCell-root-63 MuiTableCell-body-65 MuiTableCell-numeric-67\\">356</td><td class=\\"MuiTableCell-root-63 MuiTableCell-body-65 MuiTableCell-numeric-67\\">16</td><td class=\\"MuiTableCell-root-63 MuiTableCell-body-65 MuiTableCell-numeric-67\\">49</td><td class=\\"MuiTableCell-root-63 MuiTableCell-body-65 MuiTableCell-numeric-67\\">3.9</td></tr></tbody></table></div>"`;
